stl/src/StlCompareStringA.cpp

```diff
@@ -95,9 +94,8 @@ extern "C" int __cdecl __crtCompareStringA(_In_z_ LPCWSTR LocaleName, _In_ DWORD
                 return 1;
             }
 
-            for (auto cp = &cpInfo.LeadByte[0]; cp[0] && cp[1]; cp += 2) {
-                const auto ch = static_cast<unsigned char>(*lpString2);
-                if (ch >= cp[0] && ch <= cp[1]) {
+            for (auto* cp = &cpInfo.LeadByte[0]; cp[0] && cp[1]; cp += 2) {

```

Comment:
auto doesn't need that pointer. If cp was const you'd need it, but it is not. Also it isn't consistent with the previous loop which uses the same code without the pointer.